internal/cluster/cluster.go

```diff
@@ -94,36 +91,23 @@ func NewManager(cfg Config, log logrus.FieldLogger, db *gorm.DB, eventsHandler e
 	return &Manager{
 		log:             log,
 		db:              db,
-		insufficient:    NewInsufficientState(log, db, hostAPI),
-		ready:           NewReadyState(log, db),
 		installing:      NewInstallingState(log, db),
-		finalizing:      NewFinalizingState(log, db),
-		installed:       NewInstalledState(log, db),
-		error:           NewErrorState(log, db),
 		prepare:         NewPrepareForInstallation(cfg.PrepareConfig, log, db),
 		registrationAPI: NewRegistrar(log, db),
 		installationAPI: NewInstaller(log, db),
 		eventsHandler:   eventsHandler,
 		sm:              NewClusterStateMachine(th),
 		metricAPI:       metricApi,
+		rp:              newRefreshPreprocessor(log),
+		hostAPI:         hostAPI,
 	}
 }
 
 func (m *Manager) getCurrentState(status string) (StateAPI, error) {
 	switch status {
 	case "":
-	case models.ClusterStatusInsufficient:
```

Comment:
What happened to all the statues that were removed?

Reply:
All the statues that were removed from this method were moved to transitions and state machine. The only states that still using the old code are: models.ClusterStatusPreparingForInstallation, models.ClusterStatusInstalling and having a separate tasks.